### ISSUE TITLE
Suggestion: USS Freedom as a limited Airbase

### ISSUE THREAD

**Issue by AdmiralKarelia** [Wyqer/kp_liberation#287]
I've noticed that, in all my time playing Liberation, both your version and the original, that the LHD/carrier is forgotten as soon as the players have the first FOB built. This is mainly because the carrier has no actual utility other than being the respawn location for the Huron.

What about the possibility of having the Freedom act as an airbase, similar to a FOB at an airfield that has aviation facilities constructed? In this scenario, giving the Freedom limited construction ability, so it can construct aircraft like the Black Wasp, Ghosthawk, Blackfish, and Blackfoot onboard, with the appropriate resource investment of course. If possible, adding some form of repair, rearm, and refuel functions onboard the ship as well, so that it can act as a proper staging point for an airborne assault or strike mission.

It may not be feasible, but it'd be great if it is. I'd love to see the Freedom actually have a purpose in a Liberation campaign.

**Comment by amunoz20** [comment]
I was thinking about this very same thing the other day while playing a port I did myself. The map I ported Liberation to has no airfields and no flat lands suitable for planes. I was thinking that maybe something like this could be implemented:

After the first FOB gets built you can go to the USS Freedom and buy "improvements" for it. This improvements could be:

- Buy ammo box
- Buy fuel box
- Buy repair box
- Buy Plane [X plane]
- Buy Helicopter [X Helicopter]
- Buy Anti Air Measures [like the CIWIS Turrets and other things BI implemented with Jets DLC]

Each of this things will cost resources (maybe from global counter?) and can be restricted to certain number like 0/4 planes 0/3 helicopters etc... Ofcourse you should have bought the Flight control prior buying any plane or helicopter (maybe?)

Why I suggest to treat this it like "improvements"? Because this could be spawn points pre located in the carrier that will spawn the thing you are buying in the right place without letting you "drag" the thing like the normal build system does to date.

It's an interesting idea to implement IMO

**Comment by Wyqer** [comment]
Yeah, there was a similar suggestion in the past.

My points for this (mostly the same as last time):
* The Operation Base (Freedom/Chimera) is only a "spawn at an area which is far enough away from the enemy" and to provide spawnpoints for the basic vehicles.
* The focus of Liberation concerning "starting operations" is your FOBs, where you can organize your resources, build your buildings, your vehicles, etc.
* For sure the Freedom is a "nice asset" but after you've your first FOB you mostly don't need it anymore.
* Aircrafts are very expensive. So at the point that you've the needed resources, you're mostly far enough progressed to also have an airfield captured and have mostly forgotten about the carrier on the sea.
* How would you suggest to have it technically solved concerning the resources?
    * There is no "access global resource pool", as resources have locality due to the fact that they're real objects in the world.
    * So from which FOB should the resources come?
    * Or should you have to ship the resources to a resource storage on the carrier?
* What is with the maps where you don't have a carrier because there is no water? Chimera isn't and won't be placed near an airstrip (as it's a too valuable sector to capture in game)

Don't get me wrong. I'm not totally against this suggestion. I've just my concerns that it's really worth the work which would be needed to implement this. And as you can see there are still open question for which there has to be suitable solutions. I'll post this discussion as link in the Liberation discord, to get some more opinions and eyes which look at it.

**Comment by AdmiralKarelia** [comment]
I could understand the viewpoint of not allowing it to be a construction point.

Honestly, I'd be happy if it simply acted as a service station for aircraft, and saved the aircraft I leave there (the same way a FOB does). Since the FOBs can't be placed on or near a capturable point, a lot of airbase FOBs are in awkward locations, like at the end of a runway. This doesn't leave a lot of area for rearm/refuel for fixed wing aircraft, which aren't as nimble in confined spaces as helicopters.

It also presents a storage concern. If I buy a Black Wasp or a Wipeout, I'm saving it even if I'm not using it. My group might have several expensive aircraft saved up, and a lot of airfield FOBs don't lend themselves well to that, while the USS Freedom would. Of course, there's the issue of saved aircraft conflicting with the Huron and Hummingbirds that respawn on the ship.

**Comment by Applejakerie** [comment]
Personally I don't think it has to be too balanced regarding carrier vs land operation bases.

My suggestion would be that from FOB X you can see improvements for the carrier (check for carrier before showing) in the support menu. They'd be pretty costly and would include pre-placed turrets/containers for defensive and refuelling/repairing/rearming purposes. I also think the carrier should be saving any nearby aircraft that aren't the pre-placed little birds or huron. The carrier becomes operational and a nice advancement and where there is no carrier, this system can be forgotten. As a side note, maybe the land operational base should also save ground vehicles locally - for anyone who's yet to have built an FOB (waiting for a better spot) but have captured enemy vehicles they intend to use.

**Comment by Wyqer** [comment]
> My suggestion would be that from FOB X you can see improvements for the carrier

That's indeed something I haven't thought of before and could be a suitable way for something like that if it'll be considered and implemented.

Basically I can understand that using the carrier as "resupply and operation base for jets" is a immersive thing and I'm already kind of convinced. Especially with the above suggestion for adding "resupply capabilities".

But what I don't see currently is the need for any "Defence" on the carrier, as I've personally never experienced the need that the carrier should be defended in any way. It is mostly far away from any sector or airspawn point (intended, of course). From my current point of view the "add defence" would just be a decorative aspect.

**Comment by amunoz20** [comment]
What @Applejakerie said was exactly what I said. It can be considered a "nice optional" to have for más that have a carrier.

And  also what @Wyqer says is correct, there is no need to defenses for it appart from the fact that would look cool. However a smart player could use those defenses to their advantage like flying the enemy close or something... All that is up to the player..

The true request here is to make the carrier able to "store" vehicles in an effort to include it in the game as some sort of ressupply base. Or even better if theres a way like to "transfer" a plane from FOB X to the carrier once bought, or buy it directly into the carrier it would help people to get air support on scenarios where no suitable terrain is present for an "improvised airfield"

**Comment by Wyqer** [comment]
Ok let's stick this once together for the further discussion.

Current state of the suggestion is:
* Carrier should save at least jets on the deck (directly or maybe with a kind of virtual storage)
* Carrier should get a possibility to resupply jets (vanilla and ACE)
* Improvements should be able to be buyed **at a FOB and not on the carrier itself** (because locality of resources)
* Maybe a possibility to "transfer" jets to the carrier or "restock" the available jets on the carrier
* Maybe defence systems, even if it's just a cosmetic thing (still not convinced on that, to be honest)
* Stuff that I've missed, for sure ^^

All the above is possible to implement "without problems". Would need some testing and maybe it needs some other tweaking we're currently not thinking about. But basically I can imagine a basic way which wouldn't be "too much work".

Nevertheless I would like to keep this discussion ongoing. Especially as now everything which happens in this repo is also announced in the [ArmA 3 Liberation Discord Channel](https://discord.gg/bpPUU48), so I hope for some more people who may want to state their opinion in this discussion here on GitHub.

> it would help people to get air support on scenarios where no suitable terrain is present for an "improvised airfield"
Good point, I agree.

**Comment by Applejakerie** [comment]
One thing I did think of right now, when I added these turrets and 9 jets to the deck, the turrets were able to fire up to 3-5km and would target ground units and the jets would sink into the deck. The turret range could prove annoying and unfair, and rearming them would need to be considered, almost entirely sure they fixed the deck issue however.

**Comment by amunoz20** [comment]
Yeah, I was thinking about the same... If you ported the mission to a small map (like I do) buying a turret or antiair mission would completelly negate air forces of the unit (even if thats how it actually works irl, that sounds a bit TOO OP)

**Comment by glothos** [comment]
Loved the idea, landing on a carrier is always fun as a jet pilot.
